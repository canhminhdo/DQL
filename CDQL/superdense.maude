load CDQL.maude

fmod SUPERDENSE is
    pr CDQL-SEMANTICS .

    --- defining initial states
    op initQState : -> Qubit .
    eq initQState = (q[0]: |0>) (q[1]: |0>) .
endfm

in ../debug.maude .

red in SUPERDENSE : initQState |=
([ H(0) ; CX(0, 1) ; CX(0, 1) ; H(0)] (P(0, |0>) /\ P(1, |0>))) /\
([ H(0) ; CX(0, 1) ; X(0); CX(0, 1) ; H(0)] (P(0, |0>) /\ P(1, |1>))) /\
([ H(0) ; CX(0, 1) ; Z(0); CX(0, 1) ; H(0)] (P(0, |1>) /\ P(1, |0>))) /\
([ H(0) ; CX(0, 1) ; Y(0); CX(0, 1) ; H(0)] (P(0, |1>) /\ P(1, |1>))) .