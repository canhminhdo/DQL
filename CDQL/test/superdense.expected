==========================================
reduce in SUPERDENSE : initQState |= ((([H(0) ; CX(0,1) ; (Alice2 ||' Bob)]P(0,
    |1>) /\ P(1,|0>)) /\ [H(0) ; CX(0,1) ; (Alice3 ||' Bob)]P(0,|1>) /\ P(1,
    |1>)) /\ [H(0) ; CX(0,1) ; (Alice1 ||' Bob)]P(0,|0>) /\ P(1,|1>)) /\ [H(0)
    ; CX(0,1) ; (Alice0 ||' Bob)]P(0,|0>) /\ P(1,|0>) .
result JudgmentSet: emptyJS
Bye.
